File: springmvcHelloworld/src/main/java/com/wangshibiao/spring/soapService/Helloworld.java
package com.wangshibiao.spring.soapService;

import com.wangshibiao.spring.dao.UserMapper;
import com.wangshibiao.spring.model.User;
import com.wangshibiao.spring.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationContext;
import org.springframework.context.support.ClassPathXmlApplicationContext;
import org.springframework.stereotype.Service;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.web.context.support.SpringBeanAutowiringSupport;

import javax.annotation.Resource;
import javax.jws.WebMethod;
import javax.jws.WebParam;
import javax.jws.WebService;
import javax.jws.soap.SOAPBinding;

/**
 * Created by Wang Shibiao on 2016/6/27.
 */

/**
 * 生成webservice剩余代码，执行命令：wsgen -wsdl -r F:\github\springmvc\springmvcHelloworld\src\main\resources\soap -s F:\github\springmvc\springmvcHelloworld\src\main\java -classpath F:\github\springmvc\out\build\springmvcHelloworld\classes\main\ com.wangshibiao.spring.soapService.Helloworld
 * 注意：执行wsgen之前先把extends SpringBeanAutowiringSupport注掉，执行完成后再恢复，否则命令提示错误
 */

@WebService
public class Helloworld extends SpringBeanAutowiringSupport{
    @Autowired
    UserService userService;

    @WebMethod
    public String method1(String name){
        return "Hello "+name;
    }

    @WebMethod
    public User getUser(@WebParam(name = "username") String username){
        User user = userService.getUserByUserName(username);
        return user;
    }
}

File: scala/src/main/scala/HelloWorld.scala
import org.apache.commons.lang.StringUtils

/**
  * Created by Wang Shibiao on 2017/1/17.
  * 一.命令行编译、运行
  * 1. scalac -d F:\github\springmvc\out\build\scala\classes\main F:\github\springmvc\scala\src\main\scala\HelloWorld.scala
  * 2. scala -cp F:\github\springmvc\out\build\scala\classes\main HelloWorld
  * 二.gradle编译
  * gradle scala:compileScala
  * 三. 注意
  * 验证发现如下现象：
  * 代码必须经过手动编译后，才能断点调试
  */
object HelloWorld {
  def main(args: Array[String]) {
    var var1: String = "我调用的是scala方法"
    var var2: String = "我调用的是java jar的方法"
    println(var1)
    System.out.println(var2)
    var map = Model1.fun1("scala类传给java类的信息");
    println(map.get("msg"))
  }
}
